src/QsCompiler/Tests.Compiler/TestUtils/Signatures.fs

```diff
@@ -609,6 +621,195 @@ let public ClassicalControlSignatures =
     |]
     |> _MakeSignatures
 
+let private _WithTupleTypes =
+    _MakeTypeMap [| _MakeTupleType [ "Double", Double
+                                     "String", String
+                                     "Result", Result ]
+                    _MakeTupleType [ "Double", Double
+                                     "String", String ]
+                    _MakeTupleType [ "Int", Int
+                                     "Double", Double ] |]
+
+let private _WithLambdaOperation =
+    _MakeTypeMap [| "Unit => Int is Adj + Ctl",
+                    ((ResolvedType.New UnitType, ResolvedType.New Int),
+                     {
+                         Characteristics =
+                             ResolvedCharacteristics.FromProperties [ OpProperty.Adjointable
+                                                                      OpProperty.Controllable ]
+                         InferredInformation = InferredCallableInformation.NoInformation
+                     })
+                    |> QsTypeKind.Operation |]
+
+/// Expected callable signatures to be found when running Lambda Lifting tests
+let public LambdaLiftingSignatures =
+    [| // Basic Lift
+        (_DefaultTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "_Foo", [||], "Int" // The generated operation
```

Comment:
If this there is a small fix for what your asking, then please point it out and I can look into it. But I'm concerned that what you're suggesting would be a major refactor of the testing infrastructure, and if that's the case, it needs to be addressed in a separate PR as a separate work item.

Reply:
My suggestion is: In the lambda lifting tests, you look up the `QsCallable` for the original callable and the lifted lambdas. I think it would be clearer if you asserted that the `QsCallable.Signature` matched the expected type in LambdaLiftingTests.fs, rather than in Signatures.fs.

Reply:
Okay thank you for the suggestion, but unfortunately I'm going to have to reject that suggestion for this PR, as that would be too big a change to the testing infrastructure, and not particularly relevant to the purpose of this PR. If you'd like, you can file a GitHub issue for this refactor, and we can take it from there.

Reply:
Sorry, but to clarify, I think it is relevant to this PR because it seems like it prevents the lambda lifting tests from having adequate test coverage of the lifted lambda types. For example, in the "Functor Support Basic Return" test:

```qsharp
// Functor Support Basic Return
namespace Microsoft.Quantum.Testing.LambdaLifting {
    operation Foo() : Unit {
        let lambda1 = () => 0;
        let lambda2 = () => ();
    }
}
```

```fsharp
// Functor Support Basic Return
(_DefaultTypes,
 [|
     LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
     LambdaLiftingNS, "_Foo", [||], "Int"
     LambdaLiftingNS, "_Foo", [||], "Unit"
 |])
```

This asserts that there exists a lifted lambda of type `Unit => Int` and a lifted lambda of type `Unit => Unit`. But I couldn't  find any assertion that `lambda1` was specifically substituted with a lifted lambda of type `Unit => Int`, and `lambda2` was specifically substituted with a lifted lambda of type `Unit => Unit` (and not vice versa). Is this correct?

Reply:
OK, so ignoring the file organization for now, I'm just asking for this kind of assertion to be made somewhere in the test suite. I couldn't find a test case that does that, but I might have missed one.

Reply:
This should be satisfied once I've added a test for the case suggested [here](https://github.com/microsoft/qsharp-compiler/pull/1302#discussion_r791074688).

Reply:
I've added a test case that checks for this now.